### DISCUSSION TITLE
Connectivity state

### DISCUSSION THREAD

**Discussion by ciakov** [maplibre/flutter-maplibre-gl#490] [Q&A] [answered]
In order to be able to use my own storage and customizing tiles, I am using a local HttpServer to serve tiles into MapLibre.
In general it is working fine, but when there is no netrwork there is no request to my server.
In native Android (java) there was a method Mapbox.setConnected(true), 
that was overriding ConnectityManager, and everything worked fine.
Is there something equivalent in flutter-maplbre?

**Comment by josxha** [comment]
Hi @ciakov, thanks for bringing this question up. I looked into the code and found a global `setOffline` function. It should work by just calling:
```dart
setOffline(true);
```

**Reply by ciakov** [reply]
Hi @josxha, thank you for your response but it didn't solved my problem.
I have tried it, but unfortunately it does not work.
I tried setOffline(true) and setOffline(false) (just in case of invert logic),  with the same result.
If the phone has no internet connection, MapLibre does not perform any http request.
If it helps I am using local httpServer into the phone, which means that the tile urls are in the form http://localhost:[port]/....
I try adding breakpoint into the http serving function and if I am not connected to internet I am not receiving any request. If I have network conectivity everything works fine.
Please advise.

**Reply by josxha** [reply]
You're right, setConnected can be set to `false` and `null` but not to `true`.
I'm not sure what the intentions where when implementing it.
```java
boolean offline = methodCall.argument("offline");
ConnectivityReceiver.instance(context).setConnected(offline ? false : null);
```
In that case it's sadly currently not possible to set setConnected to true. But it should be possible to modify this behavior. 🤔

**Reply by ciakov** [reply]
Hi @josxha,
Can you give me any hint on how I can modify this behavior?

**Reply by josxha** [reply, marked as answer]
1. fork the repository
2. Edit the file `\maplibre_gl\android\src\main\java\org\maplibre\maplibregl\GlobalMethodHandler.java` so that setConnected can be set to true, similar on iOS.
3. Commit your changes to your fork
4. use maplibre_gl from your fork
5. consider opening a pull request

**Reply by ciakov** [reply]
@josxha , 
Thank you very much.
